### DISCUSSION TITLE
Sending an Async MQTT event

### DISCUSSION THREAD

**Discussion by trlafleur** [arendst/Tasmota#15107] [Q&A] [answered]
What is the proper way to send an MQTT message async to the normal TelePeriod time....
any examples??

thanks

**Comment by sfromis** [comment, marked as answer]
All MQTT messages are async. You can use the [Publish](https://tasmota.github.io/docs/Commands/#publish) command when you want, like from rules, where the [cookbook](https://tasmota.github.io/docs/Rules/#rule-cookbook) has several examples doing Publish.

**Comment by trlafleur** [comment]
I want to do this in a new sensor in C, so I will look for code examples...

thanks

~~ _/) ~~~~ _/) ~~~~ _/) ~~~~ _/) ~~

Tom Lafleur

**Reply by barbudor** [reply]
tasmota\xdrv_28_pcf8574.ino has some example of sending MQTT messages whenever an input pin changed state:
https://github.com/arendst/Tasmota/blob/4069bd1914041c2d155c0cdd72542a63963b0736/tasmota/xdrv_28_pcf8574.ino#L258

Or similar in MCP230XX driver:
https://github.com/arendst/Tasmota/blob/4069bd1914041c2d155c0cdd72542a63963b0736/tasmota/xsns_29_mcp230xx.ino#L360

**Reply by barbudor** [reply]
the 1st parameter define the %prefix% to use:
``` c++
enum TopicOptions { CMND, STAT, TELE, nu1, RESULT_OR_CMND, RESULT_OR_STAT, RESULT_OR_TELE };
```
The first 3 ones always use that prefix
The last 3 one use RESULT or the prefix depending on `SetOption4`
